src/wallet/walletutil.cpp

```diff
@@ -52,12 +52,16 @@ static bool IsBerkeleyBtree(const fs::path& path)
 std::vector<fs::path> ListWalletDir()
 {
     const fs::path wallet_dir = GetWalletDir();
+    const std::string base = wallet_dir.string();
     std::vector<fs::path> paths;
 
-    for (auto it = fs::recursive_directory_iterator(wallet_dir); it != end(it); ++it) {
+    for (auto it = fs::recursive_directory_iterator(wallet_dir); it != fs::recursive_directory_iterator(); ++it) {
+        assert(it->path().string().compare(0, base.size(), base) == 0);
+        const std::string relative = it->path().string().substr(base.size() + 1); // Account for directory separator
```

Comment:
It might be good to pull this logic out into a reusable `fs::path LexicallyRelative(fs::path path, fs::path base)` function. Then later we could replace it with standard [lexically_relative](https://en.cppreference.com/w/cpp/filesystem/path/lexically_normal).

Reply:
Done.